### ISSUE TITLE
Jenkinsfile: best way to derive actual PR commit

### ISSUE THREAD

**Issue by vdice** [deis/controller-sdk-go#45]
As https://ci.deis.io/job/Deis/pipeline-syntax/globals doesn't appear to expose the actual PR commit sha, we run an incantation along the lines of `git rev-parse HEAD | git log --pretty=%P -n 1 | cut -c1-40` to derive it.  (fwiw, we also have to use that first `git rev-parse HEAD` command to grab the merge commit)

We should either find the best practice `git` command fu to derive the PR commit or see about filing an issue upstream to the Jenkins Pipeline project about exposing it.  (Also, the GitHub Pull Request Plugin gives us this info when used with the jobs defined in DSL in the https://github.com/deis/jenkins-jobs repo -- we also might consider seeing if this plugin can be used in a Jenkinsfile)

**Comment by vdice** [comment]
Also note that sometimes the merge commit **is** the PR commit, which will break the logic mentioned above (when that logic is run, we will actually pass down the master commit that has been merged w/ the PR commit).  This is obviously not good as the actual PR changes will then not be tested.  Example:

```
 > git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from 2 remote Git repositories
 > git config remote.origin.url https://github.com/deis/controller-sdk-go.git # timeout=10
Fetching upstream changes from https://github.com/deis/controller-sdk-go.git
 > git --version # timeout=10
using .gitcredentials to set credentials
 > git config --local credential.username deis-admin # timeout=10
 > git config --local credential.helper store --file=/tmp/git490279147800695968.credentials # timeout=10
 > git -c core.askpass=true fetch --tags --progress https://github.com/deis/controller-sdk-go.git +refs/heads/*:refs/remotes/origin/*
 > git config --local --remove-section credential # timeout=10
 > git config remote.origin1.url https://github.com/deis/controller-sdk-go.git # timeout=10
Fetching upstream changes from https://github.com/deis/controller-sdk-go.git
using .gitcredentials to set credentials
 > git config --local credential.username deis-admin # timeout=10
 > git config --local credential.helper store --file=/tmp/git1050038001610347382.credentials # timeout=10
 > git -c core.askpass=true fetch --tags --progress https://github.com/deis/controller-sdk-go.git +refs/pull/*/head:refs/remotes/origin/pr/*
 > git config --local --remove-section credential # timeout=10
Merging master commit 616e9a45a59b5a8afa5dc34a29a79b4ab3604ddc into PR head commit 96a8fd6a0b4add3df12fa159b7da72cba9b7b963
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 96a8fd6a0b4add3df12fa159b7da72cba9b7b963
 > git merge 616e9a45a59b5a8afa5dc34a29a79b4ab3604ddc # timeout=10
 > git rev-parse HEAD^{commit} # timeout=10
Merge succeeded, producing 96a8fd6a0b4add3df12fa159b7da72cba9b7b963
Checking out Revision 96a8fd6a0b4add3df12fa159b7da72cba9b7b963 ()
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 96a8fd6a0b4add3df12fa159b7da72cba9b7b963
```
...
```
Entering stage Update local glide.yaml with controller-sdk-go repo 'github.com/vdice/controller-sdk-go' and version '616e9a45a59b5a8afa5dc34a29a79b4ab3604ddc'
```

**Comment by vdice** [comment]
Fixed by #50; closing.
